### DISCUSSION TITLE
How to add reference to files within a bucket into Db

### DISCUSSION THREAD

**Discussion by Kay2dan** [supabase/supabase#2750] [Questions] [answered]
Hi, I created a bucket and uploaded files. How can I add reference (address) of each file to a corresponding table row within Db?

**Comment by dshukertjr** [comment, marked as answer]
Hi @Kay2dan!

The easiest way of doing this if your buckets are public is though using  [getPublicUrl()](https://supabase.io/docs/reference/javascript/storage-from-getpublicurl) method to get the URL of the file, which you can store in your database. 

If your bucket is not public, then you can store the path to the file (e.g. `folder/avatar1.png`) as text in your database and use [createSignedUrl()](https://supabase.io/docs/reference/javascript/storage-from-createsignedurl) method to create a URL for authorized users to access the file.
